fenix/app/src/test/java/org/mozilla/fenix/browser/TranslationsIconBindingTest.kt

```diff
@@ -0,0 +1,168 @@
+                BrowserState(
+                    tabs = listOf(tab),
+                    selectedTabId = tabId,
+                ),
+            )
+
+            val binding = TranslationsIconBinding(
+                browserStore = browserStore,
+                sessionId = tabId,
+                onIconChanged = onIconChanged,
+            )
+            binding.start()
+
+            val detectedLanguages = DetectedLanguages(
+                documentLangTag = englishLanguage.code,
+                supportedDocumentLang = true,
+                userPreferredLangTag = spanishLanguage.code,
+            )
+
+            val translationEngineState = TranslationEngineState(
+                detectedLanguages = detectedLanguages,
+                error = null,
+                isEngineReady = true,
+                requestedTranslationPair = TranslationPair(
+                    fromLanguage = englishLanguage.code,
+                    toLanguage = spanishLanguage.code,
+                ),
+            )
+
+            val supportLanguages = TranslationSupport(
+                fromLanguages = listOf(englishLanguage),
+                toLanguages = listOf(spanishLanguage),
+            )
+
+            browserStore.dispatch(
+                TranslationsAction.SetSupportedLanguagesAction(
+                    tabId = tab.id,
+                    supportedLanguages = supportLanguages,
+                ),
+            ).joinBlocking()
+
+            browserStore.dispatch(
+                TranslationsAction.TranslateStateChangeAction(
+                    tabId = tabId,
+                    translationEngineState = translationEngineState,
+                ),
+            ).joinBlocking()
+
+            browserStore.dispatch(
+                TranslationsAction.TranslateSuccessAction(
+                    tabId = tab.id,
+                    operation = TranslationOperation.TRANSLATE,
+                ),
+            ).joinBlocking()
+
+            Mockito.verify(onIconChanged).invoke(
+                true,
+                true,
+                englishLanguage,
+                spanishLanguage,
+            )
+        }
+
+    @Test
+    fun `GIVEN translationState WHEN translation status isExpectedTranslate THEN invoke onIconChanged callback`() =
+        runTestOnMain {
+            browserStore = BrowserStore(
+                BrowserState(
+                    tabs = listOf(tab),
+                    selectedTabId = tabId,
+                ),
+            )
+
+            val binding = TranslationsIconBinding(
+                browserStore = browserStore,
+                sessionId = tabId,
+                onIconChanged = onIconChanged,
+            )
+            binding.start()
+
+            browserStore.dispatch(
+                TranslationsAction.TranslateExpectedAction(
+                    tabId = tabId,
+                ),
+            ).joinBlocking()
+
+            Mockito.verify(onIconChanged).invoke(
```

Comment:
```suggestion
           verify(onIconChanged).invoke(
```
Could you import `Mockito.verify` so we can drop `Mockito` here

Reply:
Done